src/kv/rocksdb_cache/BinnedLRUCache.cc

```diff
@@ -151,13 +151,20 @@ void BinnedLRUCacheShard::EraseUnRefEntries() {
   }
 }
 
-void BinnedLRUCacheShard::ApplyToAllCacheEntries(void (*callback)(void*, size_t),
-                                           bool thread_safe) {
+void BinnedLRUCacheShard::ApplyToAllCacheEntries(
+  const std::function<void(const rocksdb::Slice& key,
+                           void* value,
+                           size_t charge,
+                           DeleterFn)>& callback,
+  bool thread_safe)
```

Comment:
I see in rocksdb parent they removed thread_safe, does it impact our code or should we also be fine without it?
https://github.com/ceph/rocksdb/compare/c540de6f709b66efd41436694f72d6f7986a325b...6a2bb9473340846d75d32aa8f3eac91c744f92d8#diff-f222361de6f386b0531818faf1c24486fc8d7edff999250051384de6dbb1ad1dL420-L422

is it to support previous versions as well?

Reply:
yes, i plan to backport it to pacific.